oio/account/backend.py

```diff
@@ -147,20 +170,33 @@ class AccountBackend(RedisConnection):
             local damaged_objects = info[3];
             local missing_chunks = info[4];
 
-            if damaged_objects == false then
-              damaged_objects = 0;
-            end;
-            if missing_chunks == false then
-              missing_chunks = 0;
+            if not string.find(container_name, '+segments') then
+                total_objects = total_objects + objects;
             end;
+            total_bytes = total_bytes + bytes;
 
-            update_bucket_stats(
-                ckey, bkey, blistkey, account_id, container_name, bucket_name,
-                '', false,
-                objects, bytes, damaged_objects, missing_chunks);
+            if damaged_objects ~= false then
+              total_damaged_objects = total_damaged_objects + damaged_objects
+            end;
+            if missing_chunks ~= false then
+              total_missing_chunks = total_missing_chunks + missing_chunks
+            end;
           end;
+          new_marker = container_name
         end;
-        """)
+
+        -- Increment the counters.
+        redis.call('HINCRBY', bkey, 'objects', total_objects);
+        redis.call('HINCRBY', bkey, 'bytes', total_bytes);
+        redis.call('HINCRBY', bkey, 'damaged_objects',
+                    total_damaged_objects);
+        redis.call('HINCRBY', bkey, 'missing_chunks',
+                    total_missing_chunks);
+
+        redis.call('HSET', lkey, 'marker', new_marker);
+        return { new_marker };
```

Comment:
We could only return a boolean: `return { i ~= batch_size }`

Reply:
:+1: